### DISCUSSION TITLE
Is it possible to incline create new row in table?

### DISCUSSION THREAD

**Discussion by verbeckii** [cloudscape-design/components#1630] [Q&A] [answered]
Hi

I have implemented inline editing functionality for my table, and I am now looking to enhance it further. Specifically, I would like to add a "Create" button that will allow me to effortlessly insert a new empty row into the table. Once the empty row is added, I can easily make inline modifications to the required columns, and upon saving, the changes will be reflected by creating a new row on the server. Can you please assist me with this task?
Only one solution I see right now - just create new row in the server once I click "Create" button (sure if your API allow you to create that empty row, but most time its not)
![image](https://github.com/cloudscape-design/components/assets/60220767/9092f478-9370-4eb9-9204-b79d081e77ac)

**Comment by johannes-weber** [comment, marked as answer]
Hi @verbeckii, in theory, your approach could work by first adding an empty row which can then be inline-edited.
Nevertheless there are some implications on that approach: how do you handle scenarios, where sorting, filtering or pagination is actively used. E.g. if the user has sorted a table and is currently on page x/y - would the specific position in the table be considered when adding a new row? 

You could use a [Modal](https://cloudscape.design/components/modal/?tabId=playground&example=with-text-input) for your use-case (assuming a group has two fields, name and store). The modal opens on the same page and contains the form for creating new Groups. Find more information about the [Create resources pattern on our website](https://cloudscape.design/patterns/resource-management/create/).
